cypress/integration/transforms_spec.js

```diff
@@ -0,0 +1,238 @@
+      localStorage.setItem("home:welcome:show", true);
+
+      // Go to sample data page_size
+      cy.visit(`${Cypress.env("opensearch_dashboards")}/app/home#/tutorial_directory/sampleData`);
+
+      // Click on "Sample data" tab
+      cy.contains("Sample data", { timeout: 20000 }).click({ force: true });
+      // Load sample eCommerce data
+      cy.get(`button[data-test-subj="addSampleDataSetecommerce"]`).click({ force: true });
+
+      // Verify that sample data is add by checking toast notification
+      cy.contains("Sample eCommerce orders installed", { timeout: 60000 });
+
+      // Visit ISM Transforms Dashboard
+      cy.visit(`${Cypress.env("opensearch_dashboards")}/app/${PLUGIN_NAME}#/transforms`);
+
+      // Common text to wait for to confirm page loaded, give up to 60 seconds for initial load
+      cy.contains("Create transform", { timeout: 60000 });
+    });
+
+    describe("can be created", () => {
+      it("successfully", () => {
+        // Confirm we loaded empty state
+        cy.contains(
+          "Transform jobs help you create a materialized view on top of existing data."
+        );
+
+        // Route to create transform page
+        cy.contains("Create transform").click({ force: true });
+
+        // Type in transform ID
+        cy.get(`input[placeholder="my-transformjob1"]`).type(TRANSFORM_ID, { force: true });
+
+        // Get description input box
+        cy.get(`textarea[data-test-subj="description"]`).focus().type("some description");
+
+        // Enter source index
+        cy.get(`div[data-test-subj="sourceIndexCombobox"]`)
+          .find(`input[data-test-subj="comboBoxSearchInput"]`)
+          .focus()
+          .type("opensearch_dashboards_sample_data_ecommerce{enter}");
+
+        // Enter target index
+        cy.get(`div[data-test-subj="targetIndexCombobox"]`)
+          .find(`input[data-test-subj="comboBoxSearchInput"]`)
+          .focus()
+          .type("target_index{enter}");
+
+        // Click the next button
+        cy.get("button").contains("Next").click({ force: true });
+
+        // Confirm that we got to step 2 of creation page
+        cy.contains("Select fields to transform");
+
+        // Setup Group and aggregation
+        cy.wait(2000);
+        cy.contains("category.keyword").parent().parent().parent().next()
+          .children().first().children().first().children().first()
+          .click({ force: true });
+
+        cy.contains("Group by terms").click({ force: true });
+
+        // Confirm group was added
+        cy.contains("category.keyword_terms");
+
+        // Add aggregable field
+        cy.contains("50 columns hidden").click({ force: true });
+        cy.contains("taxless_total_price").click({ force: true });
+        // Click out of the window
+        cy.contains("Select fields to transform").click({ force: true });
+
+        cy.contains("taxless_total_price").parent().parent().parent().next()
```

Comment:
Are we not able to locate the correct element here so there's so many layers of parent() and next()? If we don't have a more straightforward way to access the correct element, can you add a comment to let others know what each layer means?

Reply:
Correct. The only part of the header that has info about which header it is, is the actual text in the header itself. We have to find that, navigate up to the div that contains that text, go to the next div that contains the button, and go back down to the button and click it. Added a comment to explain.